Modules/SubModule_Import/xASL_imp_NII2BIDS_SubjectSessionRun.m

```diff
@@ -298,6 +298,25 @@
             end
         end
     end
+    
+    
+    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    % Check for FME Hadamard sequences
+    if isfield(jsonLocal,'SeriesDescription')
+        isHadamardFME = ~isempty(regexp(char(jsonLocal.SeriesDescription),'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
+        if isHadamardFME
+            startDetails = regexp(char(jsonLocal.SeriesDescription),'\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once');
+            hadamard.type = xASL_str2num(jsonLocal.SeriesDescription(startDetails:startDetails+1),'auto');
```

Comment:
I do like structuring, but they don't understand the concept of it with BIDS ;)
So we better come up now with names that have a chance to be adopted in BIDS - with a bit of luck, we won't have to rename stuff later.

I would thus change hadamard.type to HadamardType and add that directly to JSON.

Similarly NumberOfTEs or even better NumberTEs should be used with full words.Same for TI, except that we use the term PLD. These two actually in theory don't need to be under Hadamard, because they can have other use. In theory. Because in practice, this information is duplicitous to what is in EchoTime and PLD fields. So these two fields will never make it to BIDS. So for now, lets make our life easier and use the fields `HadamardNumberTI` and `HadamardNumberTE`. Later on, we will make sure that the correct values are properly saved to the standard fields and these fields are no longer neceary. What do you think?

Reply:
4da6aec 👍